src/main/java/by/exadel/internship/service/InternshipService.java

```diff
@@ -0,0 +1,40 @@
+package by.exadel.internship.service;
+
+
+import by.exadel.internship.dto.internshipDTO.GuestInternshipDTO;
+import by.exadel.internship.entity.Internship;
+import by.exadel.internship.exception.NoEntityException;
+import by.exadel.internship.mapper.InternshipMapper;
+import by.exadel.internship.repository.InternshipRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.UUID;
+
+@Service
+@RequiredArgsConstructor
+public class InternshipService {
+
+    private final InternshipRepository internshipRepository;
+    private final InternshipMapper internShipMapper;
+
+    public GuestInternshipDTO getById(UUID uuid) {
+
+        Internship internship = internshipRepository
+                .findById(uuid)
+                .orElseThrow(() -> new NoEntityException("Entity was not found!"));
```

Comment:
Please, do more specific message:
```suggestion
                .orElseThrow(() -> new NoEntityException(String.format("Intership with id %s was not found!", uuid)));
```

Reply:
done